assets/javascripts/discourse/widgets/discourse-reactions-actions.js.es6

```diff
@@ -245,21 +245,68 @@ export default createWidget("discourse-reactions-actions", {
             );
+            const current_user_reaction = this.attrs.post.current_user_reaction;
+            const current_user_used_main_reaction = this.attrs.post.current_user_used_main_reaction;
+            const reactions = Object.assign([], this.attrs.post.reactions);
+            const reaction_users_count = this.attrs.post.reaction_users_count;
+
             if (
               this.attrs.post.current_user_reaction &&
               this.attrs.post.current_user_reaction.id == params.reaction
             ) {
-              dropReaction(postContainer, params.reaction, () => {
-                return CustomReaction.toggle(
-                  params.postId,
-                  params.reaction
-                ).then(resolve);
-              });
+              this.collapsePanels();
+              this.dropUserReaction();
+              this.attrs.post.reaction_users_count -= 1;
+              this.attrs.post.current_user_used_main_reaction = false;
+              this.setCurrentUserReaction();
+              later(() => {
+                dropReaction(postContainer, params.reaction, () => {
+                  return CustomReaction.toggle(
+                    params.postId,
+                    params.reaction
+                  ).then((value) => {
+                    resolve;
+                    if(value == undefined) {
+                      this.attrs.post.current_user_reaction = current_user_reaction;
+                      this.attrs.post.current_user_used_main_reaction = current_user_used_main_reaction;
+                      this.attrs.post.reactions = reactions;
+                      this.attrs.post.reaction_users_count = reaction_users_count;
+                      this.scheduleRerender();
+                    }
+                  });
+                });
+              }, 100);
             } else {
+              this.dropUserReaction();
+              this.addUserReaction(params.reaction);
+
+              if (!this.attrs.post.current_user_reaction) {
+                this.attrs.post.reaction_users_count += 1;
+              }
+              this.setCurrentUserReaction(params.reaction);
+
+              if (
+                this.attrs.post.current_user_reaction &&
+                this.attrs.post.current_user_reaction.id ==
+                  this.siteSettings.discourse_reactions_like_icon
+              ) {
+                this.attrs.post.current_user_used_main_reaction = true;
+              } else {
+                this.attrs.post.current_user_used_main_reaction = false;
+              }
+
               addReaction(postContainer, params.reaction, () => {
-                CustomReaction.toggle(params.postId, params.reaction).then(
-                  resolve
-                );
+                this.collapsePanels();
+                CustomReaction.toggle(params.postId, params.reaction).then((value) => {
+                  resolve;
+                  if(value == undefined) {
```

Comment:
js uses triple ===

Reply:
also !value

Reply:
we need to check for undefined, as !value doesn't work in our case